flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/FlinkKubeClientFactory.java

```diff
@@ -92,13 +94,32 @@ public FlinkKubeClient fromConfiguration(Configuration flinkConfig, String useCa
         LOG.debug("Setting namespace of Kubernetes client to {}", namespace);
         config.setNamespace(namespace);
 
+        // This could be removed after we bump the fabric8 Kubernetes client version to 4.13.0+ or
+        // use the a shared connection for all ConfigMap watches. See FLINK-22006 for more
+        // information.
+        trySetMaxConcurrentRequest(config);
+
         final NamespacedKubernetesClient client = new DefaultKubernetesClient(config);
         final int poolSize =
                 flinkConfig.get(KubernetesConfigOptions.KUBERNETES_CLIENT_IO_EXECUTOR_POOL_SIZE);
         return new Fabric8FlinkKubeClient(
                 flinkConfig, client, createThreadPoolForAsyncIO(poolSize, useCase));
     }
 
+    @VisibleForTesting
+    static void trySetMaxConcurrentRequest(Config config) {
+        final String configuredMaxConcurrentRequests =
+                Utils.getSystemPropertyOrEnvVar(
+                        Config.KUBERNETES_MAX_CONCURRENT_REQUESTS,
+                        String.valueOf(Config.DEFAULT_MAX_CONCURRENT_REQUESTS));
```

Comment:
I am wondering whether we shouldn't rather make it configurable via a Flink configuration option? That way it would also be documented for our users and easier to use.

Reply:
Since fabric8 kubernetes-client provides many internal advanced configuration properties[1], I do not want to introduce such a special case for `kubernetes.max.concurrent.requests`. If you think it is a problem that we do not have the documentation, maybe we could add a section in `native_kubernetes.md#Flink on Kubernetes Reference`

[1]. https://github.com/fabric8io/kubernetes-client/blob/v4.9.2/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L65

Reply:
Ok, then let's add a section about how to configure the Fabric8 kube client to `native_kubernetes.md` where we state how to set these experts options and also tell explicitly about `KUBERNETES_MAX_CONCURRENT_REQUESTS` if users want to submit a lot of jobs.